### PR TITLE
Update install os job to handle more input case of installDisk param

### PR DESCRIPTION
**Pull request by WangWinson**
- set default value ```firstdisk``` or ```sda``` when driveId catalog does not exist
- throw error if installDisk is number but could not found in driveId catalog
- when installDisk is empty, use wwid of SATADOM, if not exist use the first device in driveId catalog
- set installDisk default to null in install-esx and install-centos task definition

@RackHD/corecommitters @iceiilin @sunnyqianzhang @pengz1 @cgx027 
This change is similar to previous PR https://github.com/RackHD/on-tasks/pull/86 on 1.0 branch. Also, it covers https://github.com/RackHD/RackHD/issues/53

### CODE REVIEW

File: spec/lib/jobs/install-os-job-spec.js [hunk 1]
```diff
@@ -255,5 +260,32 @@ describe('Install OS Job', function () {
...
+                return job._convertInstallDisk();
+            }).then(function() {
+                expect(job.options.installDisk).to.equal('sda');
+            });
+        });
+
+        it('should set firstdisk if installDisk is null and catalog is empty (ESXi)', function() {
+            job.options.osType = 'esx';
+            job.options.installDisk = null;
+            waterline.catalogs.findMostRecent = sinon.stub().resolves({});
```
Comment (WangWinson) [inline spec/lib/jobs/install-os-job-spec.js]: sinon is global , could be ignore

**Comment by iceiilin** [comment]
+1

File: lib/jobs/install-os.js [hunk 2]
```diff
@@ -167,32 +167,45 @@ function installOsJobFactory(
...
         }).then(function(catalog) {
-            var wwid = catalogSearch.findDriveWwidByIndex(
-                catalog.data,
-                (self.options.osType.toLowerCase() === 'esx'),
-                disk
-            );
-
-            if (!wwid) {
-                return Promise.reject(new Error('Fail to find the WWID for installDisk'));
+            var isEsx = /^esx$/i.test(self.options.osType);
```
Comment (sunnyqianzhang) [inline lib/jobs/install-os.js]: Does this mean a variable starting with 'e' and ending with 'x'?

File: lib/jobs/install-os.js [hunk 3]
```diff
@@ -233,5 +246,23 @@ function installOsJobFactory(
         });
     };
 
+     /**
+     * Search the driveid catalog and lookup the corresponding drive WWID of SATADOM
+     * @param {Object} catalog - the catalog data of drive id
```
Comment (yyscamper) [inline lib/jobs/install-os.js]: catalog is Array

File: spec/lib/jobs/install-os-job-spec.js [hunk 4]
```diff
@@ -231,9 +232,13 @@ describe('Install OS Job', function () {
             return expect(job._convertInstallDisk()).to.be.rejectedWith(Error);
         });
 
-        it('should do nothing if installDisk is not specified', function() {
+        it('should set SATADOM wwid as default if installDisk is not specified', function() {
             job.options.installDisk = null;
-            return expect(job._convertInstallDisk()).to.not.be.rejected;
+            return Promise.resolve().then(function() {
```
Comment (yyscamper) [inline spec/lib/jobs/install-os-job-spec.js]: Since now _convertInstallDisk always return promise, we can remove this line in point of simplicity.

**Comment by yyscamper** [comment]
- the task definition of install centos and install exs should be updated to default set installDisk as null or undefined, otherwise the feature of finding default SATADom will not work well.

- it's better to change the commit message, MAG-xx and ODR-xxx is used internally, it's not used in open source, meanwhile, only write the issue number doesn't give clear message what you have done in this PR

File: lib/jobs/install-os.js [hunk 5]
```diff
@@ -167,32 +167,45 @@ function installOsJobFactory(
...
 
-        //If disk is empty then do nothing.
         //If disk is string, it means user directly input the drive wwid, so don't need conversion.
-        if ((!disk && disk !== 0) || _.isString(disk)) {
+        if (_.isString(disk) && !_.isEmpty(disk)) {
             return Promise.resolve(disk);
         }
 
-        if (!_.isNumber(disk)) {
+        if (!_.isNumber(disk) && !_.isEmpty(disk)) {
```
Comment (yyscamper) [inline lib/jobs/install-os.js]: it's better to have a comment here why the disk can be empty, if people is not familiar with our design, he may feel a little confused

File: lib/jobs/install-os.js (hunk 2 above)
Comment (WangWinson) [inline lib/jobs/install-os.js]: This means the ```osType``` should equals ```'esx'``` in lower case

File: lib/jobs/install-os.js (hunk 3 above)
Comment (WangWinson) [inline lib/jobs/install-os.js]: Good catch. Copy paste issue

File: spec/lib/jobs/install-os-job-spec.js (hunk 4 above)
Comment (WangWinson) [inline spec/lib/jobs/install-os-job-spec.js]: updated unit test to remove the ```Promise.resolve()```

File: lib/jobs/install-os.js (hunk 5 above)
Comment (WangWinson) [inline lib/jobs/install-os.js]: Comment added

**Comment by WangWinson** [comment]
@yyscamper , as to the commit message, I think the story/ODR number is useful for issue tracking purpose. So that people who review this PR could directly find the related story/ODR item.

I agree that it will be confused to external user what's the story/ODR number means. I think the better way is to put a related github issue number. But currently , we still using the ODR system and do not have one to one matched github issue. I bring up this topic on today's daily meeting, hopefully we could found out a better way thru discussion.

**Comment by yyscamper** [comment]
+1

**Pull request merged**